extensions/wikia/Wall/index/CommentsIndex.class.php

```diff
@@ -0,0 +1,255 @@
+<?php
+
+/**
+ * CommentsIndex Class
+ * @author Kyle Florence, Saipetch Kongkatong, Tomasz Odrobny
+ */
+
+class CommentsIndex {
+	use \Wikia\Logger\Loggable;
+
+	/** @var CommentsIndex $instance */
+	private static $instance = null;
+
+	/** @var CommentsIndexEntry[] $objectCache */
+	private $objectCache = [];
+
+	private function __construct() {
+	}
+
+	public static function singleton(): CommentsIndex {
+		if ( !( static::$instance instanceof CommentsIndex ) ) {
+			static::$instance = new CommentsIndex();
+		}
+
+		return static::$instance;
+	}
+
+	/**
+	 * Given a Comments Index Entry, update the corresponding row in DB.
+	 *
+	 * @param CommentsIndexEntry $entry
+	 * @param DatabaseBase $dbw
+	 */
+	public function updateEntry( CommentsIndexEntry $entry, DatabaseBase $dbw = null ) {
+		$dbw = $dbw ?? wfGetDB( DB_MASTER );
+
+		$entry->setLastTouched( $dbw->timestamp() );
+
+		$dbw->update(
+			'comments_index',
+			$entry->getDatabaseRepresentation(),
+			[ 'comment_id' => $entry->getCommentId() ],
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "comment_id" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)